wolfcrypt/src/asn.c

```diff
@@ -11030,15 +11075,127 @@ int wc_EccPublicKeyDecode(const byte* input, word32* inOutIdx,
     if (ret != 0)
         return ret;
 
-    /* ecc params information */
-    ret = GetObjectId(input, inOutIdx, &oidSum, oidIgnoreType, inSz);
-    if (ret != 0)
-        return ret;
+    if (input[*inOutIdx] == (ASN_SEQUENCE | ASN_CONSTRUCTED)) {
+#ifdef WOLFSSL_CUSTOM_CURVES
+        ecc_set_type* params;
+        int len;
+        char* point;
 
-    /* get curve id */
-    curve_id = wc_ecc_get_oid(oidSum, NULL, 0);
-    if (curve_id < 0)
-        return ECC_CURVE_OID_E;
+        ret = 0;
+
+        params = XMALLOC(sizeof(*params), NULL, DYNAMIC_TYPE_ECC_BUFFER);
+        if (params == NULL)
+            ret = MEMORY_E;
+
+        if (ret == 0) {
+            XMEMSET(params, 0, sizeof(*params));
+            params->name = "Custom";
+            params->id = ECC_CURVE_CUSTOM;
+
+            if (GetSequence(input, inOutIdx, &length, inSz) < 0)
+                ret = ASN_PARSE_E;
+        }
+
+        if (ret == 0) {
+            GetInteger7Bit(input, inOutIdx, inSz);
+            if (GetSequence(input, inOutIdx, &length, inSz) < 0)
+                ret = ASN_PARSE_E;
+        }
+        if (ret == 0) {
+            SkipObjectId(input, inOutIdx, inSz);
+            ret = ASNToHexString(input, inOutIdx, (char**)&params->prime, inSz,
+                                                 NULL, DYNAMIC_TYPE_ECC_BUFFER);
```

Comment:
You can pass `key->heap` into the `ASNToHexString` heap arg here and for the other XMALLOC's added. ECC for non small stack builds needs to honor the heap ptr.

Reply:
Fixed